Fix small issues related to tensors

phipf opened the pull request:
#### References to other Issues or PRs

None. My searches did not yield any corresponding issues.


#### Brief description of what is fixed or changed

1. The `_op_priority` mechanism was broken for tensors, because binary special methods were not decorated with `call_highest_priority`.
2. Tensor heads are pretty-printed.

#### Other comments

*Regarding 1.*: The decorator is not added to `__rtruediv__`, because it seems desirable to always raise an error for a division by a tensor. It is also not added to `__pow__` and `__rpow__`, because powers of objects with co- and contravariant indices seem somewhat questionable in the first place, which I guess is the reason why these methods are not (fully) implemented.

*Regarding 2.*: I considered possible confusions with tensor indices and Unicode subscripts and superscripts, but since the former are printed in the lines above and below the head and clearly have a different size, I don't see this as a problem.

#### Release Notes

<!-- BEGIN RELEASE NOTES -->

* tensor
  * Fixed the `_op_priority` mechanism for `TensExpr`.

* printing
  * Enabled pretty-printing for tensor heads.

<!-- END RELEASE NOTES -->

phipf committed:
fix: _op_priority mechanism for TensExpr

TensExpr set _op_priority but failed to decorate its special methods
with call_highest_priority.

phipf committed:
author: Add phipf to .mailmap

phipf committed:
Pretty-print tensor heads

Before, a tensor head with a name like "p_0" would be printed as is, and
not as "p₀". This negatively affected readability.

oscarbenjamin commented on .mailmap:
```diff
@@ -1747,6 +1747,7 @@ oittaa <8972248+oittaa@users.noreply.github.com>
 omahs <73983677+omahs@users.noreply.github.com>
 pekochun <hamburg_hamburger2000@yahoo.co.jp>
 philwillnyc <56197213+philwillnyc@users.noreply.github.com>
+phipf <phipf@online.de>
```
Is this the name that you want to have recorded in the AUTHORS file?

phipf replied:
Yes, that is correct.

phipf committed:
fix: mypy complaining about incompatible inherited methods

This problem was introduced in commit
c6bd1553f2eeafbdddffbac13cdde3b8ce09444d.

phipf committed:
test: Pretty-printing of tensor heads

This tests the changes from commit
163ffd108d117a34e4d1a14195c65921c5830f83.

phipf commented on sympy/printing/pretty/pretty.py:
```diff
@@ -1269,7 +1269,7 @@ def _print_NDimArray(self, expr):
 
     def _printer_tensor_indices(self, name, indices, index_map={}):
-        center = stringPict(name)
+        center = stringPict(pretty_symbol(name))
```
Sorry, I considered this change too trivial for a test. Should have looked in `/sympy/printing/pretty/tests/test_pretty.py` first.

phipf replied:
Do I need to change anything else or can I consider this resolved? I appologize if you meant to take a look later and I'm just too impatient, but based on your previous fast reactions I suspect I might have misunderstood something about what changes you expect.
